dashboard-app/src/dashboard/ByTime/datePickerConstraints.ts

```diff
@@ -0,0 +1,67 @@
+import moment from 'moment';
+import { DateRangePickerConstraint } from '../../components/DatePicker/types';
+
+// Date rules:
+// all:
+// 1. no future dates for either from or to
+// 2. from < to
+// 3. from >= 01 Jan 2017
+//
+// time:
+// 7. month picker
+// 8. min diff (to - from) = 0
+// 9. max diff (to - from) = 11
+// 10. default from: 11 months ago
+// 11. default to: now
+//
+// Time:
+// - min from: 01 Jan 2017       (3)
+// - max from: now               (1)
+//
+// - min to: 01 Jan 2017         (3)
+// - max to: now                 (1)
+//
+// - default from: 11 months ago (10)
+// - default to: now             (11)
+//
+// - validate:
+//   - from:
+//     - from < to || from       (2, 8)
+//     - to - from <= 11 || from (9)
+//   - to:
+//     - from < to || from       (2, 8)
+//     - to - from <= 11 || from + 11 months (9)
+
+const MIN_DATE = moment('2017-01-01');
```

Comment:
Do we wanna pull this out somewhere?

Reply:
Maybe, idk. Where?

Reply:
`dashboard-app/src/components/DatePicker/types.ts`?